### PR TITLE
Dialog to select proposal/directory when starting GUI 

### PR DESCRIPTION
**Pull request by takluyver**
image

I haven't tried to do any refactoring to take full advantage of this yet.

### CODE REVIEW

File: damnit/gui/open_dialog.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        self.ui.buttonBox.button(QDialogButtonBox.StandardButton.Ok).setEnabled(valid)
+
+    def browse_for_folder(self):
+        path = QFileDialog.getExistingDirectory()
+        if path:
+            self.ui.folder_edit.setText(path)
+
+    def get_chosen_dir(self):
+        if self.ui.proposal_rb.isChecked():
+            return Path(self.proposal_dir, "usr/Shared/amore")
```
Comment (JamesWrigley) [inline damnit/gui/open_dialog.py:64]: Nitpicking:
```suggestion
            return Path(self.proposal_dir) / "usr/Shared/amore"
```

**Review by JamesWrigley — CHANGES_REQUESTED** [review]
Things we discussed:
- Adding tests
- Open dialog from menu entry
- Creating a backend from the dialog if one doesn't exist

**Comment by takluyver** [comment]
Thanks @JamesWrigley for the review. I've made all the changes we discussed now; do you want to take another look overt the latest changes? The main change, other than the tests, is to the code for creating a DAMNIT DB and starting the backend if they don't already exist.

**Comment by JamesWrigley** [comment]
Hmmm, now I'm getting crashes when testing this on Maxwell and my local machine :thinking: Both with selecting by proposal and directory:
```
(amore) ➜  mid-2832 git:(open-db-dialog) amore-proto gui --no-kafka    
QThread: Destroyed while thread is still running
[1]    3384071 IOT instruction (core dumped)  amore-proto gui --no-kafka
```

**Comment by takluyver** [comment]
Gah, threads. I saw those errors while running the tests and fixed them there, but then didn't try again with the actual application. I'll take a look.

**Comment by takluyver** [comment]
OK, I think I've finally got the application, the local tests and the CI all working :sweat_smile: :crossed_fingers: .

**Review by JamesWrigley — APPROVED** [review]
I didn't have quite enough time to test it myself, but I trust you and CI :grin: LGTM!

(P.S. I might also suggest squashing some of the commits before merging, but I leave that up to you)

**Pull request merged**